artiq/frontend/aqctl_corelog.py

```diff
@@ -50,26 +58,29 @@ async def get_logs(host):
     writer.write(struct.pack("B", Request.PullLog.value))
     await writer.drain()
 
-    while True:
-        length, = struct.unpack(endian + "l", await reader.readexactly(4))
-        log = await reader.readexactly(length)
-
-        for line in log.decode("utf-8").splitlines():
-            m = re.match(r"^\[.+?\] (TRACE|DEBUG| INFO| WARN|ERROR)\((.+?)\): (.+)$", line)
-            levelname = m.group(1)
-            if levelname == 'TRACE':
-                level = logging.TRACE
-            elif levelname == 'DEBUG':
-                level = logging.DEBUG
-            elif levelname == ' INFO':
-                level = logging.INFO
-            elif levelname == ' WARN':
-                level = logging.WARN
-            elif levelname == 'ERROR':
-                level = logging.ERROR
-            name = 'firmware.' + m.group(2).replace('::', '.')
-            text = m.group(3)
-            log_with_name(name, level, text)
+    try:
+        while True:
+            length, = struct.unpack(endian + "l", await reader.readexactly(4))
+            log = await reader.readexactly(length)
+
+            for line in log.decode("utf-8").splitlines():
+                m = re.match(r"^\[.+?\] (TRACE|DEBUG| INFO| WARN|ERROR)\((.+?)\): (.+)$", line)
+                levelname = m.group(1)
+                if levelname == 'TRACE':
+                    level = logging.TRACE
+                elif levelname == 'DEBUG':
+                    level = logging.DEBUG
+                elif levelname == ' INFO':
+                    level = logging.INFO
+                elif levelname == ' WARN':
+                    level = logging.WARN
+                elif levelname == 'ERROR':
+                    level = logging.ERROR
+                name = 'firmware.' + m.group(2).replace('::', '.')
+                text = m.group(3)
+                log_with_name(name, level, text)
+    except:
+        pass
```

Comment:
This is overly broad (both in terms of exceptions caught and span of the ``try`` block) and masks important exceptions. Even simple bugs like a typo ``level = logging.INFIO`` in the ``try`` block become problematic with such code.

Besides, connection failures are actual errors that should be reported.